lib/phlex/kit.rb

```diff
@@ -1,31 +1,39 @@
 # frozen_string_literal: true
 
 module Phlex::Kit
-	def self.extended(mod)
-		warn "⚠️ [WARNING] Phlex::Kit is experimental and may be removed from future versions of Phlex."
-		super
+	module LazyLoader
+		def method_missing(name, *args, **kwargs, &block)
+			if name[0] == name[0].upcase && __phlex_kit_constants__.include?(name) && __get_phlex_kit_constant__(name) && methods.include?(name)
+				public_send(name, *args, **kwargs, &block)
+			else
+				super
+			end
+		end
+
+		def respond_to_missing?(name, include_private = false)
+			if name[0] == name[0].upcase && __phlex_kit_constants__.include?(name) && __get_phlex_kit_constant__(name) && methods.include?(name)
+				true
+			else
+				super
+			end
+		end
 	end
 
-	# When a kit is included in a module, we need to load all of its components.
-	def included(mod)
-		constants.each { |c| const_get(c) if autoload?(c) }
-		super
+	include LazyLoader
+
+	def self.extended(mod)
+		warn "⚠️ [WARNING] Phlex::Kit is experimental and may be removed from future versions of Phlex."
+		mod.include(LazyLoader)
```

Comment:
The reason to include this as a module rather than just define `method_missing` and `respond_to_missing?` directly like below is so it can call `super`.